cpp/core/jni/JniWrapper.cc

```diff
@@ -688,14 +688,14 @@ JNIEXPORT jlong JNICALL Java_io_glutenproject_vectorized_ShuffleWriterJniWrapper
     jstring data_file_jstr,
     jint num_sub_dirs,
     jstring local_dirs_jstr,
-    jboolean prefer_spill,
+    jboolean prefer_evict,
     jlong allocator_id,
     jboolean write_schema,
     jlong firstBatchHandle,
     jlong taskAttemptId,
     jint push_buffer_max_size,
     jobject celeborn_partition_pusher,
-    jstring shuffle_writer_type_jstr) {
+    jstring partition_writer_type_jstr) {
   JNI_METHOD_START
```

Comment:
I am feeling that we can avoid sharing the same JNI entrance method when creating these two types of partition writers. 

Then we can remove the partition-writer-specific options from `SplitOptions` (I think it supposed to be `ShuffleWriterOptions`), then we can passing a "partition writer factory" into the shuffle writer initialization routine to avoid choosing partition writer implementation from there.

It's OK to keep the code as it is and to do the refactors sometime later.

Reply:
@zhztheplayer  OK. I will refactor this part after this pr is merged.